### PR TITLE
Implementing range header used with revision selection

### LINKED ISSUE #3479: AzConfig Implement Revision Range Selection
Add the capability for revisions to select a specified range

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #3479 

Implements the range header functionality for revision selection.

### CODE REVIEW

File: applicationconfig/client/src/main/java/com/azure/applicationconfig/models/SettingSelector.java [hunk 1]
```diff
@@ -175,4 +176,23 @@ public SettingSelector fields(SettingFields... fields) {
...
         return this;
     }
+
+    /**
+     * Gets the {@link Range} used to select a specific range of revisions with {@code listSettingRevisions}.
+     * If {@code null}, the service returns all revisions.
+     * @return The {@link Range} used to select a range of revisions.
+     */
+    public Range range() {
+        return (range == null) ? null : new Range(range.start(), range.end());
```
Comment (conniey) [inline applicationconfig/client/src/main/java/com/azure/applicationconfig/models/SettingSelector.java]: Was there an error when we tried to return the range object itself? I wonder why we had to construct a new one if Range is an immutable object.

File: applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationAsyncClientTest.java [hunk 2]
```diff
@@ -618,6 +619,34 @@ public void listRevisions() {
                 .verifyComplete();
     }
 
+    /**
+     * Verifies that the range header for revision selections returns the expected values.
+     */
+    @Test
+    public void listRevisionsWithRange() {
```
Comment (conniey) [inline applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationAsyncClientTest.java]: Is there a test for 416? (RangeNotSatisfiable)

File: applicationconfig/client/src/main/java/com/azure/applicationconfig/models/SettingSelector.java (hunk 1 above)
Comment (alzimmermsft) [reply applicationconfig/client/src/main/java/com/azure/applicationconfig/models/SettingSelector.java]: Changed to return just the range value

File: applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationAsyncClientTest.java (hunk 2 above)
Comment (alzimmermsft) [reply applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationAsyncClientTest.java]: Added

**Pull request merged**